utils/map.go

```diff
@@ -43,3 +44,42 @@ func GetKeysToDelete(tags map[string]string, upstreamTags map[string]string) []*
 	}
 	return updateUntags
 }
+
+func GetInstanceTags(templateTags []*ec2.LaunchTemplateTagSpecification) map[string]*string {
+	tags := make(map[string]*string)
+
+	for _, tag := range templateTags {
+		if aws.StringValue(tag.ResourceType) == ec2.ResourceTypeInstance {
+			for _, t := range tag.Tags {
+				tags[aws.StringValue(t.Key)] = t.Value
+			}
+		}
+	}
+	return tags
+}
+
+func CreateTagSpecs(instanceTags map[string]*string) []*ec2.LaunchTemplateTagSpecificationRequest {
+	if len(instanceTags) == 0 {
+		return nil
+	}
+
+	tags := make([]*ec2.Tag, 0)
+	for key, value := range instanceTags {
+		tags = append(tags, &ec2.Tag{Key: aws.String(key), Value: value})
+	}
+	return []*ec2.LaunchTemplateTagSpecificationRequest{
+		{
+			ResourceType: aws.String(ec2.ResourceTypeInstance),
+			Tags:         tags,
+		},
+	}
+}
+
+func CompareStringMaps(map1, map2 map[string]string) bool {
+	for key, value1 := range map1 {
+		if value2, exists := map2[key]; !exists || value1 != value2 {
+			return false
+		}
+	}
+	return len(map1) == len(map2)
```

Comment:
nit: can just return false early if this is not true and skip iterating. Probably doesn't actually matter for resource tags since there can only be 50 I believe.

Reply:
Done.